modules/data.land/R/extract_soil_nc.R

```diff
@@ -31,168 +25,147 @@
 #' @export
 #'  
 extract_soil_gssurgo <- function(outdir, lat, lon, size=1, grid_size=3, grid_spacing=100, depths=c(0.15,0.30,0.60)){
```

Comment:
With the change back to a set distance instead of a grid, I think it would also make sense to revert the `grid_size` and `grid_spacing` parameters back to something more like the `radius` that was here before #3534.

Reply:
This can also greatly simplify aoi setup -- lines 30 through 42 below could reduce to
```
aoi <- data.frame(lon = lon, lat = lat) |>
  terra::vect(crs = "epsg:4326") |>
  terra::buffer(radius)
```

Reply:
Followup thought: I'd also love to have support for passing in a custom area of interest so that I could retrieve e.g. soil properties for one whole field instead of a fixed radius/grid that approximates it. I think the change above would make that pretty easy -- if the user provides a lat/lon then construct AOI as shown here, if they provide a polygon then pass that straight into `mukey.wcs`.

Does the following (!untested!) design make sense to you?

```suggestion
extract_soil_gssurgo <- function(outdir, lat, lon, aoi = NULL, size = 1, radius = 500, depths=c(0.15, 0.30, 0.60)) {
  if (missing(aoi)) {
    aoi <- data.frame(lon = lon, lat = lat) |>
      terra::vect(crs = "epsg:4326") |>
      terra::buffer(radius)
  }
```

... along with deletion of current lines 30 to 42 below?

Reply:
i don't really have a strong preference here radius vs grid. Switching to `radius` makes more sense since we are no longer doing grid sampling (single `fetchSDA` call)
implemented changes, thanks!